src/sdk/pynni/nni/compression/torch/utils/shape_dependency.py

```diff
@@ -0,0 +1,191 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import csv
+import logging
+
+from nni._graph_utils import TorchModuleGraph
+
+
+CONV_TYPE = 'aten::_convolution'
+ADD_TYPES = ['aten::add', 'aten::add_']
+CAT_TYPE = 'aten::cat'
+logger = logging.getLogger('Shape_Dependency')
+
+
+class ChannelDependency:
+    def __init__(self, model=None, dummy_input=None, traced_model=None):
+        """
+        This model analyze the channel dependencis between the conv
+        layers in a model.
+
+        Parameters
+        ----------
+        model : torch.nn.Module
+            The model to be analyzed.
+        data : torch.Tensor
+            The example input data to trace the network architecture.
+        traced_model : torch._C.Graph
+            if we alreay has the traced graph of the target model, we donnot
+            need to trace the model again.
+        """
+        # check if the input is legal
+        if traced_model is None:
+            # user should provide model & dummy_input to trace the model or a already traced model
+            assert model is not None and dummy_input is not None
+        self.graph = TorchModuleGraph(model, dummy_input, traced_model)
+        self.dependency = dict()
+        self.build_channel_dependency()
+
+    def get_parent_convs(self, node):
```

Comment:
do we support linear layers?

Reply:
Do you mean channel dependencies introduced by the linear layers?

Reply:
Yes.

Reply:
Linear usually will not introduce channel dependency, because it is used at the end of the model in most cases. By the way, does the linear layer have the concept of output channel？If so, we should add the support for the Linear layers.

Reply:
I think `output_features` in linear layers is similar of output channel. But you're right, linear layers usually not introduce channel dependency.

Reply:
It cost negligible effort to support the Linear at the same time, so, why not? Thanks for the advice!